waiter/src/waiter/async_request.clj

```diff
@@ -160,11 +160,16 @@
 
 (defn route-params->uri
   "Converts the route params to a uri.
-   The function expects prefix to end with a slash and location to begin with a slash.
-   Returns a formatted url: prefix/{request-id}/{router-id}/{service-id}/{host}/{port}{location}"
-  [prefix {:keys [host location port request-id router-id service-id]}]
-  (let [encode #(if %1 (URLEncoder/encode %1 "UTF-8") (str %1))]
-    (str prefix (encode request-id) "/" (encode router-id) "/" service-id "/" (str host) "/" (str port) location)))
+   The function expects location to begin with a slash.
+   Returns a uri in one of the following formats (depending on whether :proto is set):
+   - /waiter-async/action/{request-id}/{router-id}/{service-id}/{host}/{port}{location}
+   - /waiter-async/v2/action/{request-id}/{router-id}/{service-id}/{host}/{port}/{proto}{location}"
```

Comment:
Should we change the `v2` version of the url to accept an encrypted dictionary like we once discussed? This may prevent the need for a `v3` version of the url later.

Reply:
Done